### ISSUE TITLE
Der angegebene Schlüssel war nicht im Wörterbuch angegeben

### ISSUE THREAD

**Issue by websystemsoftware** [2sic/2sxc#1561]
Hi,
I have DNN 9.02.00

I Installed 2sxc 9.30 with the OWL ImageSlider App

every 2 days my Images in OWL Slider on the Homepage are missing!
and the i can see only the following Error Message:
----------------------------------------------------------
AbsoluteURL:
DefaultDataProvider:DotNetNuke.Data.SqlDataProvider, DotNetNuke
ExceptionGUID:8e3ac4f6-67c1-4139-8742-59b582ae9435
AssemblyVersion:
PortalId:-1
UserId:-1
TabId:-1
RawUrl:
Referrer:
UserAgent:
ExceptionHash:W5XtDytnaIGfUGsoYnEZ/kkexT4=
Message:Search: Error while indexing module 613 on tab 59, portal 0
StackTrace:
bei ToSic.SexyContent.Environment.Dnn7.DnnBusinessController.GetModifiedSearchDocuments(ModuleInfo moduleInfo, DateTime beginDate) in C:\Projects\2sxc-dnn742\Website\DesktopModules\ToSIC_SexyContent\2sxc Dnn\Environment\Dnn7\DnnBusinessController.cs:Zeile 89. bei DotNetNuke.Services.Search.ModuleIndexer.IndexSearchDocuments(Int32 portalId, ScheduleHistoryItem schedule, DateTime startDateLocal, Action`1 indexer)
InnerMessage:Der angegebene Schlüssel war nicht im Wörterbuch angegeben.

InnerStackTrace:
bei System.ThrowHelper.ThrowKeyNotFoundException() bei System.Collections.Generic.Dictionary`2.get_Item(TKey key) bei ToSic.SexyContent.ContentGroup.get_Template() in C:\Projects\2sxc-dnn742\Website\DesktopModules\ToSIC_SexyContent\ToSic.Sxc\SexyContent\ContentGroup.cs:Zeile 70. bei ToSic.SexyContent.ContentBlocks.ModuleContentBlock..ctor(IInstanceInfo instanceInfo, Log parentLog, ITenant tenant, IEnumerable`1 overrideParams) in C:\Projects\2sxc-dnn742\Website\DesktopModules\ToSIC_SexyContent\ToSic.Sxc\SexyContent\ContentBlocks\ModuleContentBlock.cs:Zeile 84. bei ToSic.SexyContent.Environment.Dnn7.Search.SearchController.GetModifiedSearchDocuments(IInstanceInfo instance, DateTime beginDate) in C:\Projects\2sxc-dnn742\Website\DesktopModules\ToSIC_SexyContent\2sxc Dnn\Search\SearchController.cs:Zeile 53. bei ToSic.SexyContent.Environment.Dnn7.DnnBusinessController.GetModifiedSearchDocuments(ModuleInfo moduleInfo, DateTime beginDate) in C:\Projects\2sxc-dnn742\Website\DesktopModules\ToSIC_SexyContent\2sxc Dnn\Environment\Dnn7\DnnBusinessController.cs:Zeile 85.
Source:ToSic.Sxc.Dnn
FileName:
FileLineNumber:0
FileColumnNumber:0
Method:
-----------------------------------------------------------
Fehler: Img1 ist zur Zeit nicht verfügbar. DotNetNuke.Services.Exceptions.ModuleLoadException: Der angegebene Schlüssel war nicht im Wörterbuch angegeben. ---> System.Collections.Generic.KeyNotFoundException: Der angegebene Schlüssel war nicht im Wörterbuch angegeben. bei System.ThrowHelper.ThrowKeyNotFoundException() bei System.Collections.Generic.Dictionary`2.get_Item(TKey key) bei ToSic.SexyContent.ContentGroup.get_Template() in C:\Projects\2sxc-dnn742\Website\DesktopModules\ToSIC_SexyContent\ToSic.Sxc\SexyContent\ContentGroup.cs:Zeile 70. bei ToSic.SexyContent.ContentBlocks.ModuleContentBlock..ctor(IInstanceInfo instanceInfo, Log parentLog, ITenant tenant, IEnumerable`1 overrideParams) in C:\Projects\2sxc-dnn742\Website\DesktopModules\ToSIC_SexyContent\ToSic.Sxc\SexyContent\ContentBlocks\ModuleContentBlock.cs:Zeile 84. bei ToSic.SexyContent.View.get_SxcInstance() in C:\Projects\2sxc-dnn742\Website\DesktopModules\ToSIC_SexyContent\View.ascx.cs:Zeile 23. bei ToSic.SexyContent.View.Page_PreRender(Object sender, EventArgs e) in C:\Projects\2sxc-dnn742\Website\DesktopModules\ToSIC_SexyContent\View.ascx.cs:Zeile 52. --- Ende der internen Ausnahmestapelüberwachung ---

----------------------------------------------------------------
**aftere this error i must place a new OWL Slider App on the Side and fill with Images.....
2 days later the Images again gone from allone and i see only the Error Message again!

this must have something todo with the schedduler  

what can i do ??**

Thanks
Benny
![2sxc-error](https://user-images.githubusercontent.com/19758307/39872318-e1b77168-5467-11e8-8ec9-187a6a6aae80.JPG)

**Comment by Eudoxe** [comment]
I had this problem and post this : 
https://stackoverflow.com/questions/50143009/dnn-indexer-fails-with-2sxc-content-modules

I don't know which of my actions was the solution but I don't have the problem anymore. And I don't want to touch anything right now it's working.

Good luck!

**Comment by iJungleboy** [comment]
Thanks for posting this. I currently have no idea what it could be - so I would need more infos to follow up on this.

**Comment by iJungleboy** [comment]
For example: can you force the issue to appear, by starting the search index? or does it go away with a DNN restart?

**Comment by Eudoxe** [comment]
I ran into the problem after 1 week without it. It SEEMS related to the recycling of the application pool.

One week ago, I set the automatic recycling to one week. This time, I just unchecked unchecked it. I will see if there is an improvement. 

I also checked in the eventviewer of the server and I discovered that the path of the temp folder for the cached images in the web.config was not existing in the file system. I recreated the subdirectory without changing the web.config entry. 

I still don't understand why and how all of this is interrelated...

**Comment by iJungleboy** [comment]
so let's keep this open until you have a better idea what the cause is, or someone else has a suggestion.

**Comment by iJungleboy** [comment]
I believe this may be related to #1564. We're working on it, but don't know the cause yet.

**Comment by iJungleboy** [comment]
I have a few guesses. Does it always happen on the same app, and only that app? What is the App-Id of that app, and what is your portal ID?

**Comment by websystemsoftware** [comment]
Hi in my Case was the problem with the owl slider app

but at the moment i cannot say something...
i removed the app and use some other module until the problem is fixed...
else my client becomes crazy when always the images gone from alone and he see only errors

the problem was the owl slider app workes some time and then all the images was gone from allone
and i see only a error message...

**Comment by websystemsoftware** [comment]
but i will install the app on a hidden site for test and give you a feedback

**Comment by iJungleboy** [comment]
is the owl-app still installed, if yes, could you tell me what ID it has?

**Comment by websystemsoftware** [comment]
yes is installed..
which id you must have and where i found this id ?

**Comment by websystemsoftware** [comment]
![id](https://user-images.githubusercontent.com/19758307/40167050-4c2e676a-59c0-11e8-9bd8-ce006ea4c2db.JPG)

**Comment by websystemsoftware** [comment]
helps this ?

**Comment by iJungleboy** [comment]
not quite - I need the app-id. fastest way to see it is to shift-click on a toolbar button - the new window will have the app-id in the url.

**Comment by websystemsoftware** [comment]
hi,
appId=6

**Comment by iJungleboy** [comment]
do you have multiple portals (just trying to guess, if the ids get swapped by accident somewhere)

**Comment by websystemsoftware** [comment]
no i have only one portal

**Comment by iJungleboy** [comment]
Since we can't reliably reproduce it yet, I probably won't have a fix in 9.31, but I'm creating a deep-insights debug view to look into the data-model as is in memory to better track it down #1565. I hope we'll have that on an installation when we next see something like this...

**Comment by iJungleboy** [comment]
I believe I fixed it - the cause would have been this:

1. an initializer which only runs once loads global content-types
1. if it fails, it doesn't try again
1. it used Server.MapPath to find paths
1. If the search-index of DNN hit it first, it would fail because Server didn't exist (search runs in a background thread)

So if the first access to an app happens through a search-indexer, it will try to load some things - fail - and never try again. I changed it to use HostingEnvironment.MapPath and believe it works.
